### DISCUSSION TITLE
Problems with implot and gflush

### DISCUSSION THREAD

**Discussion by burston** [iraf-community/iraf#296] [Q&A] [answered]
I have found differences between IRAF 2.16.1 and 2.17.
This concerns the display when running implot (also identify, others).
I am running a Windows 10 host, Oracle’s VMBox 7.0.6 with Linux Mintmate 21 as guest. I  installed 2.17.1 via the Mintmate Software Manager which ran flawlessly.
The differences of concern noted are:
1. In 2.17 the frame is named gterm-iraf. In 2.16 this is named irafterm. Is this significant?
2. In 2.17 the word   unity   appears at the very top-left of the title block. What is this?
3. The display is all white on black background. How can I return to the colours and style that came with 2.16.1?
4. When selecting Maximise from the display drop-down list the screen is filled but the plot area is not maximised. The cursor then disappears and the “busy icon’ constantly spins – these two conditions remained after selecting Unmaximise – but were cleared when a Return key was pressed!
5. When moving the cursor on the display it is possible to inadvertently draw ‘shapes’ or to leave  ‘marks’. Despite multiple gflush commands these flaws remained at next use - regardless of task.
I have spent large amounts of time trolling through forums and the internet but have not yet found any fixes for the above. Any assistance will be appreciated.
Regards,
Patrick.

**Comment by olebole** [comment]
You  probably didn't install xgterm, so the terminal falls back to xterm.

**Reply by burston** [reply]
Thank you,
I thought I would try installing xgterm using Software Manager but the only options available were Launch and Remove which I took to show xgterm was installed.
If so what might I do next to make xgterm functional?
Patrick.

**Reply by olebole** [reply, marked as answer]
You should then just start xgterm, and launch iraf within this window.

**Reply by burston** [reply]
Thank you.
This fixed my problem.
Which now brings me back to what I thought I had fixed concerning my Discussion 289.
Should I continue here or is there a way to reopen that closed discussion?
The problem I had there  was  loading additional packages at start up
In 2..16  additional packages were simply added to the login.cl file as for example the four lines following vo  in the extract below:
```
# List any packages you want loaded at login time, ONE PER LINE.
images          # general image operators
plot            # graphics tasks
dataio          # data conversions, import export
lists           # list processing

# The if(deftask...) is needed for V2.9 compatibility.
if (deftask ("proto"))
    proto       # prototype or ad hoc tasks

tv              # image display
utilities       # miscellaneous utilities
noao            # optical astronomy packages
vo              # Virtual Observatory tools

imred
ccdred
specred
twodspec
```
I cannot find a way to do this with 2.17.
Patrick.

**Reply by olebole** [reply]
What happens if you do that with version 2.17?

**Reply by burston** [reply]
In the 2.17 version loaded via Software Manager I could find no login.cl file.
IRAF appears in my Mint Menu (Applications/Other) and from its Launcher  Properties the Command  is shown as:  /usr/bin/irafcl -x  which is a shell script file.  So I don't know from where it picks up login.cl type data.
Patrick.

**Reply by olebole** [reply]
The global login file is `/etc/iraf/login.cl`. If you want to have a user specific one, copy it to `~/.iraf/login.cl` and edit it according to you needs. For a local one, copy it to your local directory.

**Reply by burston** [reply]
ole,
I have done precisely as detailed and there was no change.
I believe that there are two different metyods results when installing IRAf, one way is by using the Software Manager which downloads from repositories  and another is using the instructions from github which also has two sets of instructions to pick from:
https://iraf-community.github.io/install.html, and
https://github.com/iraf-community/iraf/blob/main/INSTALL.md
- which differ in content but at least do provide some good detail although requiring a lengthy compile process.
As I have been working with IRAF for more than 5 years preparing a very detailed  document which I intend to unleash on the amateur spectroscopists in the near future for their use - or not.
If I cannot provide them with a mistake-proof and clear installation of IRAF then all is lost .
I will start again and make very detailed notes of what I have done and the results.
Thank you for your assistance and patience.
Patrick.

**Reply by olebole** [reply]
If you take the release 2.17 from Github, you should follow the instructions in https://iraf-community.github.io/install.html (and/or the [INSTALL.md](https://github.com/iraf-community/iraf/blob/v2.17/INSTALL.md) document that is in the tarball). Then the login files are `login.cl`, `~/.iraf/login.cl`, as for 2.16.1.

If you use the version that comes with Mint (which is basically the Debian package), then the files are taken in the order `login.cl`, `~/.iraf/login.cl`, `/etc/iraf/login.cl`. 

If you take the latest development code from Github, you should use the documentation at https://github.com/iraf-community/iraf/blob/main/INSTALL.md (which is part of the development code). This mainly adopts the change that was done in the Debian release. It will be part of the next release 2.17.1.

**Reply by burston** [reply]
[Installing IRAF.docx](https://github.com/iraf-community/iraf/files/11411826/Installing.IRAF.docx)

**Reply by olebole** [reply]
If this comes with a question please formulate it directly in the discussion.

**Reply by burston** [reply]
Have installed iraf 2.17 -1 into Mintmate 21.1 Vera using Software manager. This is the preferred method when trying to introduce new users to IRAF.
There were two versions to choose from both indicated 2.17-1:
iraf 
iraf-noao
Both install similarly but it would be good if there was only one.
After installation IRAF appeared in the Mint Menu, complete with iraf icon. very nice.

The command to launch IRAF is: /usr/bin/irafcl –x  that when launched showed a display headed  ecl.e and containing text which included:

Visit http://iraf.net if you have questions.

*** Using global login file:  /home/local/.iraf/login.cl       [There is no such file]

====================================================
I found what I believe is the correct login.cl file at /etc/iraf/login.cl

Getting closer I think – but I do not know what should be done next.
I appreciate the assistance but this is not in my field of knowledge and I really am working in the dark here.
Regards,
Patrick.

(I just discovered another version of IRAF in Software Manager in astro-iraf -  what the !!
Excuse me but how would any layperson know what to use?)

**Reply by olebole** [reply]
**iraf** contains all core packages, **iraf-noao** is the NOAO package. However, **iraf-noao** is a recommendation for **iraf**, so if you install **iraf**, you will usually get **iraf-noao** installed as well. Usually, **iraf** is what you need to install for most purposes.

**astro-iraf** is a Debian Astro metapackage that installs *all* IRAF related packages. 

After installation, you can just use IRAF. What exactly depends on what you want to do with IRAF.

If you want to change the global login file (for all users), change `/etc/iraf/login.cl`. If you want to change the login file just for yourself, copy `/etc/iraf/login.cl` to `~/.iraf/login.cl` and edit it according to your needs. If you want to change the login file for the project just in your current working directory, copy `/etc/iraf/login.cl` to your current working directory and edit it according to your needs.

However, I would also like to ask you to consider the following: IRAF is really old software, nowadays lacks further development and and institutional support. It is mainly maintained for those who cannot switch yet. Unless there is a real strong requirement, it is a really bad idea to start with it for a new project. It would be much better to use modern Python projects like [Astropy](https://astropy.org/) instead whenever it is suitable. This is much more future-proof than learning an outdated, 40 year old legacy software system.

**Reply by burston** [reply]
Ole,
Thank you.
Of course you are right. IRAF is old.
As for Astropy - I will say no more.
I have managed to install a working IRAF 2.17(1) but I cannot get it working as nicely as 2.16 -it needs tweaking and this has been something I have not managed to do.
For processing stellar spectra using long slit spectrographs in the visual there is nothing as comprehensive, as well documented and with a wealth of information from years of activity as IRAF.
I will leave this matter for now and remain a steadfast IRAF "true believer" using 2.16 of which I have a good understanding.
Regards to you and all those github IRAF supporters.
Regards,
Patrick Watson
